### PR TITLE
Added TaskGroup property support to BackgroundTaskBuilder API

### PR DESCRIPTION
**Pull request by godlytalias**
TaskGroup property support has been added to the BackgroundTaskBuilder API.

### CODE REVIEW

**Comment by AjitSurana** [comment]
@godlytalias, please elaborate the purpose of the TaskGroup property.

**Comment by godlytalias** [comment]
> @godlytalias, please elaborate the purpose of the TaskGroup property.

Its used for grouping the background tasks. So that apps can get all the tasks under that particular group using AllTasks API (https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.background.backgroundtaskregistrationgroup.alltasks?view=winrt-26100)

File: dev/BackgroundTask/BackgroundTaskBuilder/BackgroundTaskBuilder.idl [hunk 1]
```diff
@@ -24,6 +24,7 @@ namespace Microsoft.Windows.ApplicationModel.Background
             );
 
         String Name{ set; get; };
+        Windows.ApplicationModel.Background.BackgroundTaskRegistrationGroup TaskGroup{ set; get; };
```
Comment (Hemantxk) [inline dev/BackgroundTask/BackgroundTaskBuilder/BackgroundTaskBuilder.idl:27]: Are we going to expose other properties as well from UWP's BackgroundTaskBuilder? Like we have exposed AddCondition method but do not allow modifications to CancelOnConditionLoss property. What's the plan for other remaining properties.

File: specs/BackgroundTask/BackgroundTaskBuilder.md [hunk 2]
```diff
@@ -105,6 +105,7 @@ namespace Microsoft.Windows.ApplicationModel.Background
             );
 
         String Name{ set; get; };
+        Windows.ApplicationModel.Background.BackgroundTaskRegistrationGroup TaskGroup { set; get; };
```
Comment (Hemantxk) [inline specs/BackgroundTask/BackgroundTaskBuilder.md:108]: Do we need another round of API review for this, or the prior one is sufficient.

File: dev/BackgroundTask/BackgroundTaskBuilder/BackgroundTaskBuilder.idl (hunk 1 above)
Comment (godlytalias) [reply dev/BackgroundTask/BackgroundTaskBuilder/BackgroundTaskBuilder.idl:27]: CancelOnConditionLoss will not help as the win32 process won't be killed if condition is lost. If required, apps can have listeners and close win32 process if required. Other properties are also not meaningful for COM BG Task.

File: specs/BackgroundTask/BackgroundTaskBuilder.md (hunk 2 above)
Comment (godlytalias) [reply specs/BackgroundTask/BackgroundTaskBuilder.md:108]: This one was in spec earlier also, just implementation was missing

**Comment by AjitSurana** [comment]
@godlytalias, One follow up question - is it possible the same background task registration group has both Windows and Microsoft namespace background tasks? If so, AllTasks sort of properties return correct value?

**Comment by godlytalias** [comment]
> @godlytalias, One follow up question - is it possible the same background task registration group has both Windows and Microsoft namespace background tasks? If so, AllTasks sort of properties return correct value?

It returns tasks registered by both namespace APIs as internally WinAppSDK API is calling platform SDK API only. Id, Name etc.. of the tasks will be the same

**Review by AjitSurana — APPROVED** [review]

**Review by Hemantxk — APPROVED** [review]

**Pull request merged**